### ISSUE TITLE
Converting tfDist into meters

### ISSUE THREAD

**Issue by andyk735** [budryerson/TFMini-Plus#13]
Hi Bud, 

Using your example code, I would like to convert the tfDist value, which is outputted in centimeters, into meters. I have tried performing math using a double: 

`if ( tfmP.getData( tfDist, tfFlux, tfTemp)) // Get data from the device.`
   ` {`
     ` double tfMeter;`
     ` tfMeter = tfDist / 100.0;`
     ` printf( " Dist: %f", tfMeter);         // Display the distance.`
     ` break;                                   // Escape this sub-loop`
    ` }`

However, it always outputs something like:

`Loop:01 Dist: ?`
`Loop:02 Dist: ?`
`Loop:03 Dist: ?`
`Loop:04 Dist: ?`
`Loop:05 Dist: ?`

Do you have any idea of how I could solve this issue? I am not sure, it may be because tfDist is a uint16_t integer type, but I am not sure how I would fix it regardless. Hope you can help, thank you.

**Comment by budryerson** [comment]
Hey AndyK,
Try recasting tfDist as a floating point number:  `tfMeter = (float)tfDist / 100.0;`
Let me know what happens.
Bud

**Comment by andyk735** [comment]
Yep, it worked! Thank sorry for not responding sooner. 

I ran into another issue though. I am using an Arduino Uno and am trying to use the sensor with a button as well. I use Software Serial and included mySerial to get the TFMini-Plus working, because I did not know how to get it to work any other way. However, for some reason, this makes my button act weirdly, making it simply not act as a button. It's either always on or off. 
Any help?
